src/Sarif/Visitors/UpdateIndicesVisitor.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis.Sarif.Visitors
+{
+    public class UpdateIndicesVisitor : SarifRewritingVisitor
+    {
+        private IDictionary<string, int> _fullyQualifiedNameToIndexMap;
+
+        public UpdateIndicesVisitor(IDictionary<string, int> fullyQualifiedNameToIndexMap)
+        {
+            _fullyQualifiedNameToIndexMap = fullyQualifiedNameToIndexMap;
+        }
+
+        public override Location VisitLocation(Location node)
+        {
+            int index;
+
+            if (_fullyQualifiedNameToIndexMap != null && !string.IsNullOrEmpty(node.FullyQualifiedLogicalName))
```

Comment:
>IsNullOrEmpty [](start = 65, length = 13)

`NullOrEmpty` -- seems reasonable, but there is a PREfast test file where I commented that you actually had a logical location with an empty name.

Reply:
s/be cleaned up by other work. i don't think we'll want to serialize those empty values.

---
In reply to: [243858172](https://github.com/Microsoft/sarif-sdk/pull/1170#discussion_r243858172) [](ancestors = 243858172)